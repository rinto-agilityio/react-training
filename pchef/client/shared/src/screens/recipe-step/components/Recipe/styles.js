// Libs
import { StyleSheet } from 'react-native'

// Themes
import { COLORS, METRICS, FONTS } from '../../../../themes'

const tinyPadding = 3

const styles = StyleSheet.create({
  wrapper: {
    display: 'flex',
  },

  // Title
  title: {
    color: COLORS.black,
    fontWeight: 'bold',
    paddingLeft: METRICS.padding.lg,
    paddingRight: METRICS.padding.lg,
  },
  smallTitle: {
    fontSize: FONTS.fontSize.medium,
  },
  mediumTitle: {
    fontSize: FONTS.fontSize.large,
  },
  largeTitle: {
    fontSize: FONTS.fontSize.extraLarge,
  },

  // Title step
  titleStep: {
    color: COLORS.recipe.stepTitle,
    fontWeight: 'bold',
    backgroundColor: COLORS.recipe.stepTitleBg,
    paddingLeft: METRICS.padding.lg,
    paddingRight: METRICS.padding.lg,
  },

  // Description
  description: {
    color: COLORS.black,
    marginTop: METRICS.margin.md,
    paddingLeft: METRICS.padding.lg,
    paddingRight: METRICS.padding.lg,
  },
  smallDescription: {
    fontSize: FONTS.fontSize.small,
  },
  mediumDescription: {
    fontSize: FONTS.fontSize.medium,
  },
  largeDescription: {
    fontSize: FONTS.fontSize.large,
  },

  // Image
  image: {
    width: '100%',
    marginTop: METRICS.margin.md,
    justifyContent: 'space-between',
  },
  smallImage: {
    height: METRICS.bgImage.sm,
  },
  mediumImage: {
    height: METRICS.bgImage.md,
  },
  largeImage: {
    height: METRICS.bgImage.lg,
  },

  // Step
  progressStep: {
    marginRight: tinyPadding,
    padding: 0,
    minWidth: 0,
  },
  activeStep: {
    backgroundColor: COLORS.recipe.stepActive,
  },
  inactiveStep: {
    backgroundColor: COLORS.recipe.stepInActive,
  },
  smallStep: {
    width: 25,
    height: METRICS.padding.sm,
  },
  mediumStep: {
    width: METRICS.padding.xl,
    height: METRICS.padding.sm,
  },
  largeStep: {
    width: 35,
    height: METRICS.padding.sm,
  },

  // views
  views: {
    textTransform: 'uppercase',
    color: COLORS.baseGray,
    paddingLeft: METRICS.padding.lg,
    paddingRight: METRICS.padding.lg,
  },
  smallViews: {
    fontSize: METRICS.position.sm,
  },
  mediumViews: {
    fontSize: 12,
  },
  largeViews: {
    fontSize: FONTS.fontSize.small,
  },
})

export default styles
