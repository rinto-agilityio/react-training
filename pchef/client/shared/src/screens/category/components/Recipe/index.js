// @flow
// add flow above to fix for using flow with React.memo

// Libs
import React, { memo } from 'react'
import { View, Text } from 'react-native'

// Commons
import Wrapper from '../../../../layout/Wrapper'
import Icon from '../../../../components/Icon'
import Image from '../../../../components/Image'

// Constants and Helpers
import { formatStringToArray } from '../../../../helpers/utils'
import { SEPARATOR_SPLIT_STRING, ELLIPSIS_LINE, DEFAULT_IMAGE } from '../../../../constants/index'

// Themes
import { COLORS, FONTS } from '../../../../themes'

// Styles
import styles from './styles'

type Props = {
  recipe: {
    id: string,
    title: string,
    imgUrl: string,
    description: string,
  },
  isGrid: boolean,
  isFavorite: boolean,
  onPressIcon?: (id: string) => void | Promise<void>,
  size?: string,
  color?: string,
  handlePressImage?: () => void,
  customIconStyle: Object,
}

const Recipe = ({
  recipe,
  onPressIcon = () => {},
  size = '',
  color,
  handlePressImage,
  isGrid,
  isFavorite,
  customIconStyle,
}: Props) => {
  const { id, title, imgUrl, description } = recipe
  const formatDescription =
    formatStringToArray(description, SEPARATOR_SPLIT_STRING) || []

  return (
    <View style={[styles.recipe, isGrid ? styles[`${size}GridContent`] : styles[`${size}ListContent`]]}>
      <Text
        numberOfLines={ELLIPSIS_LINE}
        ellipsizeMode="tail"
        style={[styles.title, styles[`${size}Title`]]}
      > {title}
      </Text>
      <Wrapper direction="row" childPosition="left">
        <Image
          url={imgUrl || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
          handleTouch={handlePressImage}
        />
        <Wrapper
          direction="column"
          childPosition="middle"
          customStyles={[styles.wrapper, isGrid ? styles[`${size}GridWrapper`] : styles[`${size}Wrapper`]]}
        >
          {/* ingredient description */}
          <View style={styles.wrapperText}>
            {formatDescription.map((item, index) => (
              <Text
                numberOfLines={ELLIPSIS_LINE}
                ellipsizeMode="tail"
                key={index}
                style={styles[`${size}Text`]}
              >
                {item}
              </Text>
            ))}
          </View>

          <Icon
            wrapperIconStyle={[styles.icon, customIconStyle]}
            name="favorite"
            size={
              size === 'large'
                ? FONTS.fontSize.extraLarge
                : FONTS.fontSize.medium
            }
            onPress={() => onPressIcon(id)}
            color={isFavorite ? COLORS.red : color}
          />
        </Wrapper>
      </Wrapper>
    </View>
  )
}

Recipe.defaultProps = {
  onPressIcon: () => {},
  size: 'medium',
  color: COLORS.grayDarker,
  handlePressImage: () => {},
}

export default memo<Props>(Recipe)
