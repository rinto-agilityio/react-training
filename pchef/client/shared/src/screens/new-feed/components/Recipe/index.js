// @flow
// add flow above to fix for using flow with React.memo

// Libs
import React, { memo } from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from './styles'

// Components
import Image from '../../../../components/Image'
import Reaction from '../../../../components/Reaction'

// Helpers
import {
  checkContainField,
  formatFiledOnObject,
  checkContain,
} from '../../../../helpers/utils'

// Constants
import { DEFAULT_IMAGE } from '../../../../constants'

import type { RecipeType } from '../../../../types'

type Props = {
  recipe: RecipeType,
  size?: string,
  favoriteRecipe: Array<{
    id: string,
  }>,
  userToggleRecipe: (
    recipeId: string,
    favoriteRecipe: Array<{ id: string }>
  ) => Promise<{ data: { userToggleRecipe: { results: Array<string> } } }>,
  handleClickRecipe: (recipeId: string) => void,
  userToggleVote: (
    recipeId: string,
    votes: Array<string>,
    userId: string
  ) => Promise<{ data: { userToggleVote: { results: Array<string>}}}>,
  userId: string,
  wrapperIconStyle: Object,
}

const Recipe = ({
  recipe,
  size = 'large',
  favoriteRecipe,
  userToggleRecipe,
  handleClickRecipe,
  userToggleVote,
  userId,
  wrapperIconStyle,
}: Props) => {
  const { id, title, description, thumbnail, votes } = recipe

  const isFavorited = checkContainField(favoriteRecipe, id)

  const handleSaveRecipe = async () => {
    await userToggleRecipe(id, favoriteRecipe).then(({ data }) => {
      const {
        userToggleRecipe: { results },
      } = data

      if (results) {
        checkContainField(formatFiledOnObject(results), id)
      }
    })
  }

  const handleToggleVote = async () => {
    await userToggleVote(id, votes, userId)
      .then(({ data }) => {
        const {
          userToggleVote: { results },
        } = data

        if (results) {
          checkContain(votes, id)
        }
      })
  }

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View style={styles.recipe}>
        <Text
          style={[styles.title, styles[`${size}Title`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
          onPress={() => handleClickRecipe(recipe.id)}
        >
          {title}
        </Text>
        <Text
          style={[styles.description, styles[`${size}Description`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {description}
        </Text>
        <Image
          url={thumbnail || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
          handleTouch={() => handleClickRecipe(recipe.id)}
        />
      </View>
      <Reaction
        votes={votes}
        size={size}
        isFavorited={isFavorited}
        onPressFavorite={handleSaveRecipe}
        onPressVote={handleToggleVote}
        isVote={checkContain(votes, userId)}
        wrapperIconStyle={wrapperIconStyle}
      />
    </View>
  )
}

Recipe.defaultProps = {
  size: 'large',
}

export default memo<Props>(Recipe)
