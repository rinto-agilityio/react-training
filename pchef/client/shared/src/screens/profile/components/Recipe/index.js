// Libs
import React from 'react'
import { Text, View, TouchableOpacity } from 'react-native'

// Styles
import styles from './styles'

// Components
import Image from '../../../../components/Image'
import Reaction from '../../../../components/Reaction'

// Helpers
import { checkContainField, checkContain } from '../../../../helpers/utils'

// Constanst
import { DEFAULT_IMAGE } from '../../../../constants'

type Props = {
  recipe: {
    id: string,
    title: string,
    description: string,
    imgUrl: string,
    votes: Array<string>,
  },
  size?: string,
  favoriteRecipe: Array<{
    id: string,
  }>,
  userToggleRecipe: (
    recipeId: string,
    favoriteRecipe: Array<{
      id: string,
    }>
  ) => Promise<{ data: { userToggleRecipe: { results: Array<string> } } }>,
  userToggleVote: (
    recipeId: string,
    votes: Array<string>,
    userId: string
  ) => Promise<{ data: { userToggleVote: { results: Array<string>}}}>,
  userId: string,
}

const Recipe = ({
  recipe,
  size = 'large',
  favoriteRecipe,
  userToggleRecipe,
  userToggleVote,
  userId,
}: Props) => {
  const { id, title, description, imgUrl, votes } = recipe

  // Call mutation for save recipe
  const handleSaveRecipe = () => {
    userToggleRecipe(id, favoriteRecipe)
  }

  const handleToggleVote = async () => {
    await userToggleVote(id, votes, userId)
      .then(({ data }) => {
        const {
          userToggleVote: { results },
        } = data

        if (results) {
          checkContain(votes, id)
        }
      })
  }

  const isFavorited = checkContainField(favoriteRecipe, id)

  return (
    <TouchableOpacity style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View style={styles.recipe}>
        <Text
          style={[styles.title, styles[`${size}Title`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {title}
        </Text>
        <Text
          style={[styles.description, styles[`${size}Description`]]}
          numberOfLines={1}
          ellipsizeMode="tail"
        >
          {description}
        </Text>
        <Image
          url={imgUrl || DEFAULT_IMAGE}
          customImageStyle={[styles.image, styles[`${size}Image`]]}
        />
      </View>
      <Reaction
        votes={votes}
        size={size}
        isFavorited={isFavorited}
        isVote={checkContain(votes, userId)}
        onPressFavorite={handleSaveRecipe}
        onPressVote={handleToggleVote}
      />
    </TouchableOpacity>
  )
}

Recipe.defaultProps = {
  size: 'large',
}

export default Recipe
