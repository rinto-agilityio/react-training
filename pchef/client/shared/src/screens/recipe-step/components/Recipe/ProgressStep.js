// Libs
import React from 'react'

// Styles
import styles from './styles'

// Themes
import { COLORS, FONTS } from '../../../../themes'

// Components
import Button from '../../../../components/Button'
import Wrapper from '../../../../layout/Wrapper'
import Icon from '../../../../components/Icon'

type Props = {
  steps: Array<{
    step: number
  }>,
  size: string,
  step: number,
  onPressStep?: () => void,
  onPressSelectStep?: () => void,
  customStepBtn?: {},
}

const chevronIcon = (
  name: string,
  size: string,
  disabled: boolean,
  onPress: () => void
) => (
  <Icon
    name={name}
    size={FONTS.fontSize[`${size}`]}
    color={disabled ? COLORS.baseGray : COLORS.baseBlue}
    onPress={onPress}
    disabled={disabled}
    customIconStyles={{
      margin: 0,
    }}
  />
)

const Progress = ({
  steps = [],
  size = 'large',
  step,
  onPressStep,
  customStepBtn,
  onPressSelectStep,
}: Props) => {
  // Disabled prev/next icon if current step is first/last step
  const disablePrevIcon = step === 1
  const disabledNextIcon = steps.length === step

  return (
    <Wrapper direction="row">
      {chevronIcon('chevron-left', size, disablePrevIcon, onPressSelectStep)}
      <Wrapper direction="row">
        {steps.map(item => {
          const status = step === item.step ? 'active' : 'inactive'

          return (
            <Button
              key={item.step}
              onPress={onPressStep}
              buttonStyle={[
                styles.progressStep,
                styles[`${status}Step`],
                customStepBtn,
              ]}
              contentStyle={styles[`${size}Step`]}
              title=""
            />
          )
        })}
      </Wrapper>
      {chevronIcon('chevron-right', size, disabledNextIcon, onPressSelectStep)}
    </Wrapper>
  )
}

Progress.defaultProps = {
  onPressStep: () => {},
  customStepBtn: {},
  onPressSelectStep: () => {},
}

export default Progress
