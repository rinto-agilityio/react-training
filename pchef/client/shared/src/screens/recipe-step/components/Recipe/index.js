// Libs
import React, { useEffect, useState } from 'react'
import { Text, View } from 'react-native'

// Styles
import styles from './styles'

// Themes
import { COLORS, METRICS } from '../../../../themes'

// Components
import Progress from './ProgressStep'
import ImageBackground from '../../../../components/ImageBackground'
import Reaction from '../../../../components/Reaction'
import Comment from '../../../recipe/components/Comment'
import Loading from '../../../../components/Loading'
import Error from '../../../../components/Error'

// mock data
import { recipes } from '../../../../mocks'

// utils
import { findStep, compareStep, customError } from '../../../../helpers/utils'

type Props = {
  recipeSteps: Array<{
    description: string,
    imgUrl: string,
    step: number,
    title: string,
  }>,
  recipe: {
    title: string,
    votes: Array<number>,
    subTitle: string,
    views: number,
    userId: number,
    steps: Array<{
      step: number
    }>,
  },
  size: string,
  onPress?: () => void,
  customRecipe?: {},
  customTitle?: {},
  customDescription?: {},
  customImage?: {},
  customSubTitle?: {},
  customTitleStep?: {},
  loading: boolean,
  error: string
}

const Recipe = ({
  size = 'large',
  onPress,
  customRecipe,
  customTitle,
  customDescription,
  customImage = {},
  customSubTitle,
  customTitleStep,
  recipeSteps = [{
    description: '',
    imgUrl: '',
    step: 1,
    title: '',
  }],
  loading,
  error,
}: Props) => {
  // order recipeSteps by step asc
  const orderRecipeSteps = recipeSteps.sort(compareStep)
  const defaultStepInfo = orderRecipeSteps[0]
  const [stepInfo, setStepInfo] = useState()

  useEffect(() => (
    setStepInfo(defaultStepInfo)
  ), [loading, defaultStepInfo])

  const {
    title,
    subTitle,
    votes,
    userId,
    views,
  } = recipes[0]

  if (loading) {
    return <Loading />
  }

  if (error) {
    return <Error message={customError(error.graphQLErrors)} />
  }
  /**
   * Handle when user click prev or next icon
   * param {name}
   */
  const onPressSelectStep = name => {
    let nextStepInfo

    switch (name) {
      case 'next':
        nextStepInfo = findStep(orderRecipeSteps, stepInfo.step + 1)
        break;
      case 'prev':
        nextStepInfo = findStep(orderRecipeSteps, stepInfo.step - 1)
        break;
      default:
        break;
    }
    setStepInfo(nextStepInfo)
  }

  /**
   * Handle When user click any step
   * @param {step}
   */
  const onPressStep = step => {
    const stepInfoSelect = findStep(orderRecipeSteps, step)
    setStepInfo(stepInfoSelect)
  }

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <View
        style={[
          styles.recipe,
          customRecipe,
        ]}
      >
        <Text
          style={[
            styles.title,
            styles[`${size}Title`],
            customTitle,
          ]}
          onPress={onPress}
        >
          {title}
        </Text>
        <Text
          style={[
            styles.title,
            styles[`${size}Description`],
            customSubTitle,
          ]}
          onPress={onPress}
        >
          {subTitle}
        </Text>
        <ImageBackground
          url={stepInfo.imgUrl}
          customImageBg={[
            styles.image,
            styles[`${size}Image`],
            customImage,
          ]}
          resizeMode="cover"
        >
          <Text>
            <Text
              style={[
                styles.titleStep,
                styles[`${size}Description`],
                customTitleStep,
              ]}
            >
              {stepInfo.title}
            </Text>
          </Text>
          <Progress
            steps={orderRecipeSteps}
            size={size}
            step={stepInfo.step}
            onPressSelectStep={onPressSelectStep}
            onPressStep={onPressStep}
          />
        </ImageBackground>
      </View>
      <Text
        style={[
          styles.description,
          styles[`${size}Description`],
          customDescription,
        ]}
      >
        {stepInfo.description}
      </Text>
      <Reaction
        votes={votes}
        size={size}
        isFavorited={false}
      />
      <Comment
        name={`by ${userId}`}
        publishedAt={Date.now()}
        isGetTime
        customStyle={{
          paddingLeft: METRICS.largePadding,
        }}
        customNameStyles={{
          fontWeight: 'bold',
        }}
        customContentStyles={{
          fontSize: 12,
          color: COLORS.baseGray,
        }}
      />
      <Text
        style={[
          styles.views,
          styles[`${size}Views`],
        ]}
      >
        {`viewed by ${views}`}
      </Text>
    </View>
  )
}

Recipe.defaultProps = {
  onPress: () => {},
  customRecipe: {},
  customTitle: {},
  customDescription: {},
  customSubTitle: {},
  customTitleStep: {},
  customImage: {},
}

export default Recipe
