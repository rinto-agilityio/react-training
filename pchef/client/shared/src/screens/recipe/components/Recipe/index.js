// Libs
import React, { useState } from 'react'
import { View } from 'react-native'

// Styles
import styles from './styles'

// Components
import Ingredients from './Ingredients'
import Directions from './Directions'
import Loading from '../../../../components/Loading'
import Modal from '../../../../components/Modal'
import Error from '../../components/Error'

// Helper
import { customError } from '../../../../helpers/utils'

type Props = {
  getRecipe: {
    description: string,
    id: string,
  },
  size: string,
  onSelectStep: (id: string) => void,
  loading: boolean,
  error: Object,
  recipeSteps: Array<{
    step: number,
    title: string
  }>,
  history: Object,
}

const Recipe = ({
  getRecipe = {
    description: '',
    id: '',
  },
  size = 'medium',
  onSelectStep,
  loading,
  error,
  recipeSteps = [{
    step: 1,
    title: '',
  }],
  history,
}: Props) => {
  const [visible, setVisible] = useState(true)

  if (loading) {
    return <Loading />
  }

  const handleNavigateLogin = () => {
    setVisible(false)
    history.push('/login')
  }

  if (error) {
    return (
      <Modal
        visible={visible}
        onDismiss={() => handleNavigateLogin()}
        onSubmit={() => handleNavigateLogin()}
        size={size}
      >
        <Error message={customError(error.graphQLErrors)} size="medium" />
      </Modal>
    )
  }

  const {
    description,
    id,
  } = getRecipe

  return (
    <View style={[styles.wrapper, styles[`${size}Wrapper`]]}>
      <Ingredients
        description={description}
        size={size}
      />
      <Directions
        steps={recipeSteps}
        size={size}
        onSelectStep={() => onSelectStep(id)}
      />
    </View>
  )
}

export default Recipe
