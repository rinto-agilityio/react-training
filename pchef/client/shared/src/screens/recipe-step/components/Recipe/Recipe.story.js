// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'

// Components
import Recipe from '.'

// Mocks
import { recipes } from '../../../../mocks'

const recipe = recipes[0]
storiesOf('Recipe by step', module)
  .add('small', () => (
    <Recipe
      stepInfo={recipe.steps[0]}
      recipe={recipe}
      size="small"
      onPress={action('clicked')}
    />
  ))
  .add('medium', () => (
    <Recipe
      stepInfo={recipe.steps[0]}
      recipe={recipe}
      size="medium"
      onPress={action('clicked')}
    />
  ))
  .add('large', () => (
    <Recipe
      stepInfo={recipe.steps[0]}
      recipe={recipe}
      size="large"
      onPress={action('clicked')}
    />
  ))
