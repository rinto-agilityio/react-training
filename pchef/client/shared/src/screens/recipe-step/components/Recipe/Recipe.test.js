// Components
import Recipe from '.'
import ProgressStep from './ProgressStep'

// Mocks
import { recipes, user } from '../../../../mocks'

const recipe = recipes[0]
describe('Recipe by step', () => {
  const recipeProps = {
    stepInfo: recipe.steps[0],
    votes: recipe.votes,
    recipe,
    getUser: {
      favoriteRecipe: user.favoriteRecipe,
      user,
    },
    onSelectStep: jest.fn(),
  }
  const stepProps = {
    step: 1,
    onPressStep: jest.fn(),
    onPressSelectStep: jest.fn(),
  }

  it('Renders correctly recipe commponent', () => {
    const recipeComponent = renderer.create(<Recipe {...recipeProps} />).toJSON()
    expect(recipeComponent).toMatchSnapshot()
  })

  it('Renders correctly progress steps commponent', () => {
    const progressStepComponent = renderer.create(<ProgressStep {...stepProps} />).toJSON
    expect(progressStepComponent).toMatchSnapshot()
  })

  it('Render recipe component with size medium', () => {
    const props = {
      ...recipeProps,
      size: 'medium',
    }
    const recipeComponent = renderer.create(<Recipe {...props} />).toJSON()
    expect(recipeComponent).toMatchSnapshot()
  })

  it('Render progress step component with steps array', () => {
    const props = {
      ...stepProps,
      size: 'medium',
    }
    const progressStepComponent = renderer.create(<ProgressStep {...props} />).toJSON()
    expect(progressStepComponent).toMatchSnapshot()
  })

  it('Function props of recipe component should be defined', () => {
    const props = {
      ...recipeProps,
      size: 'medium',
    }

    const recipeComponent = renderer.create(<Recipe {...props} />).toJSON()

    recipeComponent.props.onPress()
    expect(Recipe.props.onPress).toBeDefined()
  })

  it('Function props of progress step component should be defined', () => {
    const props = {
      ...stepProps,
      size: 'medium',
    }
    const ProgressStep = renderer.create(<ProgressStep {...props} />).toJSON()
    ProgressStep.props.onPressStep()
    ProgressStep.props.onPressSelectStep()
    expect(ProgressStep.props.onPressStep).toBeDefined()
    expect(ProgressStep.props.onPressSelectStep).toBeDefined()
  })
})
