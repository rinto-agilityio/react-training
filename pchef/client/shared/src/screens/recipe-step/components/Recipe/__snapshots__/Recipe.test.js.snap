// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe by step Render progress step component with size medium 1`] = `ReactWrapper {}`;

exports[`Recipe by step Render progress step component with steps array 1`] = `ReactWrapper {}`;

exports[`Recipe by step Render recipe component with size medium 1`] = `ReactWrapper {}`;

exports[`Recipe by step renders correctly progress steps commponent 1`] = `ReactWrapper {}`;

exports[`Recipe by step renders correctly recipe commponent 1`] = `ReactWrapper {}`;
